lib/api/apiUtils/object/BackendInfo.js

```diff
@@ -198,6 +218,9 @@ class BackendInfo {
         if (bucketLC) {
             return bucketLC;
         }
+        if (this.legacyLocationConstraint) {
```

Comment:
We have to be careful that for new buckets we aren't defaulting to creating them in legacy. The default for new buckets should be the endpoint and if the endpoint is not known, us-east-1.

Reply:
We don't use this function when creating a bucket. 
We use this one https://github.com/scality/S3/blob/master/lib/api/bucketPut.js#L29. So it should be fine :)